torch/_inductor/kernel/templated_attention.py

```diff
@@ -149,19 +149,21 @@ def sdpa_grid(batch_size, num_heads, num_queries, d_model, meta):
         K_block_ptr = tl.advance(K_block_ptr, (0, BLOCK_N))
         V_block_ptr = tl.advance(V_block_ptr, (BLOCK_N, 0))
 
-    # write back l and m
+    # Store output and logsumexp
     acc = acc / l_i[:, None]
-    # TODO For backward support we need to add the Logsumexp
-    # l_ptrs = L + off_hz * N_CTX + offs_m
-    # tl.store(l_ptrs, m_i + tl.math.log2(l_i))
-
     idx_z = tl.program_id(1) // H
     idx_h = tl.program_id(1) % H
     idx_m = offs_m[:, None]
     idx_d = tl.arange(0, BLOCK_DMODEL)[None, :]
+
     # TODO generalize and add proper mask support
     mask = (idx_m != -1) & (idx_d != -1)
     {{store_output(("idx_z", "idx_h", "idx_m", "idx_d"), "acc")}}
+
+    # TODO dont want to write this if we dont require grad
```

Comment:
Will try this

Reply:
added the flag but inductor has no autograd information at this point so need to think about the proper way to implement this. The writes didnt lead to much slow down, but will incur extra memory that will likely be deallocated quickly in compile since there will be no uses. Still a spike though